### PR TITLE
feat(jellyfish-api-core): invalidate- and reconsiderBlock rpc

### PR DESCRIPTION
**Pull request by kresovic-milos**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:
Fixes part of #48

#### What this PR does / why we need it:
Implemented RPC from #48:
1. blockchain.invalidateBlock
2. blockchain.reconsiderBlock

### CODE REVIEW

**Review by surangap — COMMENTED** [review]
for the codeclimate issue, @fuxingloh can help you to ignore it if refactoring to smaller sizes is not required.
other one haven't seen before. `Governance / DeFi Products (pull_request_target) ` is not in other recent PRs. Could be something new.
Can you also try adding `fixes` tag as below
`Fixes part of #48 `

**Comment by fuxingloh** [comment]
> for the codeclimate issue, @fuxingloh can help you to ignore it if refactoring to smaller sizes is not required. other one haven't seen before. `Governance / DeFi Products (pull_request_target) ` is not in other recent PRs. Could be something new. Can you also try adding `fixes` tag as below `Fixes part of #48 `

I am going to remove that workflow altogether, we don't track it in the project anymore and no reason to do that now.

File: packages/jellyfish-api-core/src/category/blockchain.ts [hunk 1]
```diff
@@ -214,6 +214,28 @@ export class Blockchain {
   async waitForBlockHeight (height: number, timeout: number = 30000): Promise<WaitBlockResult> {
     return await this.client.call('waitforblockheight', [height, timeout], 'number')
   }
+
+  /**
+   * Permanently marks a block as invalid, as if it violated a consensus rule.
+   *
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/blockchain.ts:220]:
```suggestion
```

File: packages/jellyfish-api-core/src/category/blockchain.ts [hunk 2]
```diff
@@ -214,6 +214,28 @@ export class Blockchain {
...
+   * @param {string} hash of the block to mark as invalid
+   * @return Promise<void>
+   */
+  async invalidateBlock (hash: string): Promise<void> {
+    return await this.client.call('invalidateblock', [hash], 'number')
+  }
+
+  /**
+   * Removes invalidity status of a block, its ancestors and its descendants, reconsider them for activation.
+   *
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/blockchain.ts:231]:
```suggestion
```

File: packages/jellyfish-api-core/__tests__/category/blockchain/invalidateBlock.test.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'blockhash must be of length 64 (not 8, for \'12345678\')\', code: -8, method: invalidateblock')
+  })
+
+  it('should throw an error for the non-existing valid hash', async () => {
+    const promise = testing.rpc.blockchain.invalidateBlock('x'.repeat(64))
+    await expect(promise).rejects.toThrow('RpcApiError: \'blockhash must be hexadecimal string (not \'' + 'x'.repeat(64) + '\')\', code: -8, method: invalidateblock')
+  })
+
+  it('should resolve promise with null for existing hash', async () => {
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/blockchain/invalidateBlock.test.ts:27]: Add a new test or update the existing one for the follojwing scenario.
go past block height `x` -> invalidate `x` -> check if new tip is on `x-1`

File: packages/jellyfish-api-core/__tests__/category/blockchain/reconsiderBlock.test.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'blockhash must be of length 64 (not 8, for \'12345678\')\', code: -8, method: reconsiderblock')
+  })
+
+  it('should throw an error for the non-existing valid hash', async () => {
+    const promise = testing.rpc.blockchain.reconsiderBlock('x'.repeat(64))
+    await expect(promise).rejects.toThrow('RpcApiError: \'blockhash must be hexadecimal string (not \'' + 'x'.repeat(64) + '\')\', code: -8, method: reconsiderblock')
+  })
+
+  it('should resolve promise with null for existing hash', async () => {
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/blockchain/reconsiderBlock.test.ts:27]: https://github.com/DeFiCh/jellyfish/pull/860/files#r760782109
Can `reconsiderblock` for the same block and check if it returns to the old original tip.

also can update this test to check the block height after reconsidering a valid block.

**Review by surangap — CHANGES_REQUESTED** [review]
few comments added.

**Pull request closed without merging**